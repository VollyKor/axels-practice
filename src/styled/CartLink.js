import styled from 'styled-components';
import { IoCartOutline } from 'react-icons/io5';

import { Wrapper } from './utils';

export const Cart = () => (
    <Wrapper>
        <span className="mr-1">cart</span>
        <IoCartOutline size="25px" />
        <p>1</p>
    </Wrapper>
);

export const CartWrapper = styled.div`
    position: relative;
    color: var(--cart-link);
`;
export const CartText = styled.span``;

export const CartNotification = styled.div`
    position: absolute;
    top: 0;
    right: -2px;

    width: 12px;
    height: 12px;

    color: var(--white-color);
    background-color: var(--pink-color);
    border-radius: 50%50%;

    font-size: 9px;
`;
