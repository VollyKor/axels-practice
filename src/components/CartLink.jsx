import { CartWrapper, CartText, CartNotification } from '../styled/CartLink';
import { IoCartOutline } from 'react-icons/io5';

const CartLink = ({ productsInCart = 0 }) => (
    <CartWrapper>
        <CartText className="mr-1">cart</CartText>
        <IoCartOutline size="25px" />
        <CartNotification className=" d-flex justify-content-center align-items-center">
            <span>{productsInCart}</span>
        </CartNotification>
    </CartWrapper>
);

export default CartLink;
