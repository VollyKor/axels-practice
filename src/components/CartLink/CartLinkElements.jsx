import styled from "styled-components";
import { IoCartOutline } from "react-icons/io5";

export function Cart() {
  return (
    <>
      <div>
        <span className="mr-1">cart</span>
        <IoCartOutline size="25px" /> <div>1</div>
      </div>
    </>
  );
}

export const CartWrapper = styled.div`
  position: relative;
  color: #c90ea5;
`;
export const CartText = styled.span``;
export const CartNotification = styled.div`
  position: absolute;
  top: 0;
  right: -2px;

  width: 12px;
  height: 12px;

  color: #fff;
  background-color: #ff1876;
  border-radius: 50%50%;

  font-size: 9px;
`;
