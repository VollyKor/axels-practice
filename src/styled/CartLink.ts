import styled from 'styled-components';

export const CartName = styled.span``;
export const CartQuantyty = styled.p``;

export const CartWrapper = styled.div`
    position: relative;
    color: var(--cart-link);
`;
export const CartText = styled.span``;

export const CartNotification = styled.div`
    position: absolute;
    top: 0;
    right: -2px;

    width: 12px;
    height: 12px;

    color: var(--white-color);
    background-color: var(--pink-color);
    border-radius: 50%50%;

    font-size: 9px;
`;
