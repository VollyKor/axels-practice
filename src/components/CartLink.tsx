import { CartWrapper, CartText, CartNotification } from '../styled/CartLink';
import { IoCartOutline } from 'react-icons/io5';

interface Props {
    productsInCart?: number;
}

const CartLink = ({ productsInCart = 0 }: Props): JSX.Element => (
    <CartWrapper>
        <CartText className="mr-1">cart</CartText>
        <IoCartOutline size="25px" />
        <CartNotification className=" d-flex justify-content-center align-items-center">
            <span>{productsInCart}</span>
        </CartNotification>
    </CartWrapper>
);

export default CartLink;
