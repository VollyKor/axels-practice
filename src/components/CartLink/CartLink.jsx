import { IoCartOutline } from "react-icons/io5";
import { CartWrapper, CartText, CartNotification } from "./CartLinkElements";

export default function Cart() {
  return (
    <>
      <CartWrapper>
        <CartText className="mr-1">cart</CartText>
        <IoCartOutline size="25px" />
        <CartNotification className=" d-flex justify-content-center align-items-center">
          <span>{3}</span>
        </CartNotification>
      </CartWrapper>
    </>
  );
}
