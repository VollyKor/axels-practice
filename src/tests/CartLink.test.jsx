import { shallow } from 'enzyme';

import CartLink from 'components/CartLink';

it('renders correctly', () => {
    const wrapper = shallow(<CartLink productsInCart={3} />);

    expect(wrapper).toMatchSnapshot();
});

it('render default props correctly', () => {
    const wrapper = shallow(<CartLink />);

    const element = wrapper.find('span').text();
    expect(element).toContain(`0`);
});

it('render props correctly', () => {
    const productsInCart = 3;

    const wrapper = shallow(<CartLink productsInCart={productsInCart} />);

    const element = wrapper.find('span').text();
    expect(element).toContain(`3`);
});
