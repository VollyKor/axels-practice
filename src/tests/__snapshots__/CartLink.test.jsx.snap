// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<styled.div>
  <styled.span
    className="mr-1"
  >
    cart
  </styled.span>
  <IoCartOutline
    size="25px"
  />
  <styled.div
    className=" d-flex justify-content-center align-items-center"
  >
    <span>
      3
    </span>
  </styled.div>
</styled.div>
`;
